src/RFC-0321_ProcessingForeignProposals.md

```diff
@@ -0,0 +1,108 @@
+
+## Goals
+
+This RFC describes the process of distributing and processing foreign proposals in the Tari DAN Cerberus Model
+
+Across the entire network transactions must be processed or time out. When a transaction is started on a shard, it locks up substates, preventing other transactions from completing. Therefore if a transaction is started on a shard, it 
+should complete or be aborted in a timely manner to release the resources.
+
+## Related Requests for Comment
+
+<!-- * [RFC-0111: Base Node Architecture](./RFC-0111_BaseNodeArchitecture.md) -->
+None
+
+## Glossary
+
+* Block - A second layer block, consisting of ordered commands
+* Command - Command can either be Prepare, LocalPrepared, Accept, and moves a transaction into that state.
+
+## Description
+
+To solve the above problems, we'll use reliable broadcast and process foreign evidence in order.
+
+In a local shard committee, the proposed block **must** include a counter per epoch and per shard (except for the local shard). If the proposal includes transactions that involve other shards, this counter **must** be incremented.
+
+When the proposed block becomes committed locally, the block **must** be broadcast to each involved shard that was incremented, along with evidence of being committeed (The chain of QCs must be included).
+
+To ensure this, each node in the local committee will forward this committed block, along with a 3 chain of QC's proving it was committeed.
+
+As a local committee member, when I receive a foreign proposal, if it is valid I will queue up a special command ForeignProposal(number, QC_Hash) that I must propose 
+when I am next leader (if it has not been proposed already). I also **should** request all transaction hashes that I have not seen from involved_shards for each transaction in the proposal, and add them to my mempool for execution.
+
+Before transactions in the `N+1`th foreign proposal for a shard, all transactions in the `N`th foreign proposal for that shard must be sequenced into the local chain as either a TIMEOUT or a LOCALPREPARE(TxId, ForeignShardId).
+
+When a LOCALPREPARE command is found in a local proposal, there **must** be a 
+Prepare(tx) command for that transaction in or before the current proposal.
+
+In the `TIMEOUT_TIME`` block  (e.g. 1000 blocks, but should probably be equal to committee_size * X rounds) after the ForeignProposal command has been proposed, 
+all transactions in the proposal must be sequenced, either as LOCALPREPAREs or TIMEOUTs. This is to cater for the case where a transaction was prepared by a foreign committee, but was never found on the network. This is unlikely to happen 
+in a non-malicious scenario. The more common case is that all transactions will be found and have a LOCALPREPARE, but a block at `TIMEOUT_TIME` **must** be considered invalid if there are any transactions that are not LOCALPREPARED or TIMEOUT from the foreign proposal that started the timeout.
```

Comment:
Because of the new Command ForeignProposal (or perhaps ForeignProposalReceived is a better name), the nodes in the local committee have all agreed that they have received the foreign proposal, so if this command is in the local chain, the majority acknowledge that they have the ForeignProposal in hand